external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaMicroBatchStream.scala

```diff
@@ -168,7 +174,15 @@ private[kafka010] class KafkaMicroBatchStream(
     KafkaSourceOffset(JsonUtils.partitionOffsets(json))
   }
 
-  override def commit(end: Offset): Unit = {}
+  override def commit(end: Offset): Unit = {
+    if (shouldSetCommitOffsetsOnCheckpoints) {
```

Comment:
I think this whole block should be wrapped with a `try catch log` since this is best effort and shouldn't have any effect on the processing (like `commitAsync` throws transient exception). With the actual implementation the query could stop because this part is not working.

Reply:
@gaborgsomogyi I was checking the calls and I dont see any exception thrown from my code.
I could add it just to be over-protective.